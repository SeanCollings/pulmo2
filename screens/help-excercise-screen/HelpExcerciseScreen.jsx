import React from 'react';
import { StyleSheet, Text, ScrollView } from 'react-native';

import { COLORS } from '../../constants/constants';
import options from './options';

export const helpExcerciseScreenOptions = options;

const HelpExcerciseScreen = (props) => {
  return (
    <ScrollView
      style={styles.container}
      contentContainerStyle={{ alignItems: 'center' }}
    >
      <Text style={styles.textHeading}>Excercises Page</Text>
      <Text style={{ ...styles.textContent, paddingBottom: 0 }}>
        There are three types of excercises:
      </Text>
      <Text style={{ ...styles.textContent, ...styles.textBold }}>
        Strength, Endurance, Custom
      </Text>
      <Text style={styles.textContent}>
        To preview an excercise, tap on its title.
      </Text>
      <Text style={styles.textContent}>
        An excercise can be selected by either tapping on its toggle, or
        selecting it on the preview menu.
      </Text>
      <Text style={styles.textContent}>
        Create a <Text style={styles.textBold}>Custom</Text> excercise by
        inputting the title, breath cycles, rest period and number of rounds.
        <Text style={styles.textBold}> Custom</Text> excercises are saved and
        can be editted or deleted later.
      </Text>
      <Text style={styles.textContent}></Text>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 30,
    backgroundColor: COLORS.BACKGROUND,
  },
  textHeading: {
    paddingBottom: 20,
    fontSize: 20,
    color: COLORS.SECONDARY,
    fontFamily: 'tit-regular',
  },
  textContent: {
    fontSize: 16,
    width: '80%',
    textAlign: 'center',
    paddingBottom: 20,
    color: COLORS.TEXT,
    fontFamily: 'tit-light',
  },
  textBold: {
    fontFamily: 'tit-bold',
  },
});

export default HelpExcerciseScreen;
