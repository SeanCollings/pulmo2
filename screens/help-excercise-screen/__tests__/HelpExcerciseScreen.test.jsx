import React from 'react';
import { act, create } from 'react-test-renderer';
import HelpExcerciseScreen, {
  helpExcerciseScreenOptions,
} from '../HelpExcerciseScreen';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe('HelpExcerciseScreen - unit test', () => {
  test('should render', async () => {
    await act(async () => {
      tree = create(<HelpExcerciseScreen />);
    });

    expect(tree).toMatchSnapshot();
  });

  describe('helpExcerciseScreenOptions', () => {
    test('options', () => {
      const options = helpExcerciseScreenOptions();
      expect(options).toMatchSnapshot();
    });

    test('headerTitle', () => {
      const headerTitle = helpExcerciseScreenOptions().headerTitle({
        tintColor: 'white',
      });
      expect(headerTitle).toMatchSnapshot();
    });
  });
});
