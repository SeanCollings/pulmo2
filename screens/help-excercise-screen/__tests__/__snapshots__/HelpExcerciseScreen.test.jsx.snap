// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpExcerciseScreen - unit test helpExcerciseScreenOptions headerTitle 1`] = `
<Text
  style={
    Object {
      "color": "white",
      "fontSize": 20,
    }
  }
>
  Help
</Text>
`;

exports[`HelpExcerciseScreen - unit test helpExcerciseScreenOptions options 1`] = `
Object {
  "headerTitle": [Function],
}
`;

exports[`HelpExcerciseScreen - unit test should render 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
    }
  }
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingTop": 30,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#002f56",
          "fontFamily": "tit-regular",
          "fontSize": 20,
          "paddingBottom": 20,
        }
      }
    >
      Excercises Page
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 0,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      There are three types of excercises:
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-bold",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      Strength, Endurance, Custom
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      To preview an excercise, tap on its title.
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      An excercise can be selected by either tapping on its toggle, or selecting it on the preview menu.
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      Create a
       
      <Text
        style={
          Object {
            "fontFamily": "tit-bold",
          }
        }
      >
        Custom
      </Text>
       
      excercise by inputting the title, breath cycles, rest period and number of rounds.
      <Text
        style={
          Object {
            "fontFamily": "tit-bold",
          }
        }
      >
         
        Custom
      </Text>
       
      excercises are saved and can be edited or deleted later.
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    />
  </View>
</RCTScrollView>
`;
